actionmailbox/app/controllers/action_mailbox/ingresses/amazon/sns_notification.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+require "tempfile"
+
+module ActionMailbox
+  module Ingresses
+    module Amazon
+      class SnsNotification
+        def initialize(params)
+          @params = params
+        end
+
+        def subscription_confirmed?
+          confirmation_response.code&.start_with?("2")
+        end
+
+        def verified?
+          require "aws-sdk-sns"
+          Aws::SNS::MessageVerifier.new.authentic?(params.to_json)
```

Comment:
@jeremy I raised this previously but @kaspth had an opinion on this which I responded to here: https://github.com/rails/rails/pull/39364#discussion_r430051634

Since you seem to agree with me (correct me if I am wrong) that we care about byte-identical input here I have reverted to my previous implementation - I think it's the right call.